packages/core/saved-objects/core-saved-objects-migration-server-internal/src/actions/create_index.ts

```diff
@@ -74,6 +80,28 @@ export const createIndex = ({
   > = () => {
     const aliasesObject = aliasArrayToRecord(aliases);
 
+    const indexSettings = {
+      // settings not being supported on serverless ES
+      ...(esCapabilities.serverless
+        ? {}
+        : {
+            // ES rule of thumb: shards should be several GB to 10's of GB, so
+            // Kibana is unlikely to cross that limit.
+            number_of_shards: INDEX_NUMBER_OF_SHARDS,
+            auto_expand_replicas: INDEX_AUTO_EXPAND_REPLICAS,
+            // Set an explicit refresh interval so that we don't inherit the
+            // value from incorrectly configured index templates (not required
+            // after we adopt system indices)
+            refresh_interval: '1s',
+            // Bump priority so that recovery happens before newer indices
+            priority: 10,
+          }),
```

Comment:
Adapted the `createIndex` and `cloneIndex` actions to use these new capabilites.

Note that this specific work is not necessarily tested - it will be done in a follow up - the intent of the PR was to setup the feature and wire it down to the 2 migration algorithm so that we can effectively use it.